### DISCUSSION TITLE
default high cpu usage - is it normal?

### DISCUSSION THREAD

**Discussion by hnykda** [fastapi/fastapi#8126] [Questions] [answered]
Hi, 

I noticed that even with the very default hello world, uvicorn consumes about ~40% of my 2.5GHz CPU (of single core). Is this normal/expected? From WSGI I am used to values around ~5%. I was able to replicate these on two machines.

![image](https://user-images.githubusercontent.com/2741256/59026265-42c0b000-8856-11e9-90d9-d474d77dd63a.png)

**Comment by jorgerpo** [comment, marked as answer]
Try not using the reload parameter when starting the server and tell us again.

uvicorn main:app --reload

**Comment by hnykda** [comment]
Ah, yes, without the reload parameter the usage is minimal. Sorry and thanks for a quick feedback.

**Comment by wshayes** [comment]
Tom of Starlette is very open to taking a PR to make the reload option more efficient. Right now it is brute force checking every 300ms? or so.

**Comment by tiangolo** [comment]
Thanks for the help @jorgerpo and @wshayes ! :cake: :rocket: 

And thanks @hnykda for reporting back and closing the issue. :tada:
